internal/app/admin/http/handler.go

```diff
@@ -109,52 +109,72 @@ func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc
 	}
 }
 
-// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
+func printCacheEntries(keys []string, cache cache.ReadOnlyCache, w http.ResponseWriter) {
+	resp := marshallableCache{}
+	for _, key := range keys {
+		resource, err := cache.FetchReadOnly(key)
+		if err != nil {
+			w.WriteHeader(http.StatusNotFound)
+			fmt.Fprintf(w, "no resource for key %s found in cache.\n", key)
+			continue
+		}
+		resp.Cache = append(resp.Cache, resourceToPayload(key, resource)...)
+	}
+	resourceString, err := stringify.InterfaceToString(resp)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "unable to convert resource to string.\n")
+		return
+	}
+
+	if len(resp.Cache) > 0 {
+		fmt.Fprintf(w, "%s\n", resourceString)
+	}
+
+}
+
 func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		c := orchestrator.Orchestrator.GetReadOnlyCache(*o)
-		resp := marshallableCache{}
+		var keysToPrint []string
+
+		// If wildcard suffix provided, output all cache entries that match given prefix.
 		// If no key is provided, output the entire cache.
-		if cacheKey == "" {
-			keys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
+		containsWildcardSuffix := strings.HasSuffix(cacheKey, "*")
+		if cacheKey == "" || containsWildcardSuffix {
+			// Retrieve all keys
+			allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
 			if err != nil {
 				w.WriteHeader(http.StatusInternalServerError)
 				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
 				return
 			}
 
-			for key := range keys {
-				resource, err := c.FetchReadOnly(key)
-				if err != nil {
-					continue
+			// Find keys that match prefix of wildcard
+			if containsWildcardSuffix {
+				rootCacheKeyName := strings.TrimSuffix(cacheKey, "*")
+				for potentialMatchKey := range allKeys {
+					if strings.HasPrefix(potentialMatchKey, rootCacheKeyName) {
+						keysToPrint = append(keysToPrint, potentialMatchKey)
+					}
+				}
+				if len(keysToPrint) == 0 {
+					w.WriteHeader(http.StatusNotFound)
+					fmt.Fprintf(w, "no resource for key %s found in cache.\n", cacheKey)
+					return
+				}
+			} else {
+				// Add matched keys to print slice
+				for key := range allKeys {
```

Comment:
do you think it's worth special-casing the case of `allKeys` being empty?

Reply:
If you mean creating a printed response for the empty case, I can add an output statement like "cache is empty, zero entries found". If I do that though, I feel like I should also incorporate that logic for when the URL is `/cache/*`.

Reply:
Offline, we discussed this more and decided to move forward with no output in the empty return cases as it avoids adding extra messages that may need to be filtered out. Instead, I'll explore looking into adding cases like these as debug logs in the future.